applications/ExternalSolversApplication/external_includes/feast_solver.h

```diff
@@ -498,6 +428,101 @@ class FEASTSolver: public LinearSolver<TSparseSpaceType, TDenseSpaceType,
         KRATOS_CATCH("")
     }
 
+    /**
+     * Initialize CSR matrix structure for FEAST system matrix: C = z * B - A.
+     */
+    void InitializeFEASTSystemMatrix(const SparseMatrixType& B,
+                                     const SparseMatrixType& A,
+                                     ComplexSparseMatrixType& C)
+    {
+        C.resize(B.size1(), B.size2(), false);
+
+        std::vector<std::unordered_set<std::size_t> > indices(C.size1());
+
+        // indices for row begin / end
+        C.index1_data()[0] = 0;
+        for (std::size_t i = 0; i < C.size1(); ++i)
```

Comment:
just to comment that here you have a candidate for parallelization. 

you could add to c[i] the local count and use a "scan" at the end to make the incremental sum.